src/test/fifotest.cpp

```diff
@@ -0,0 +1,182 @@
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <random>
+
+#include "util/fifo.h"
+
+namespace {
+
+class FifoTest : public testing::Test {
+};
+
+TEST_F(FifoTest, writeAvailableTest) {
+    std::vector<float> data(1024);
+    FIFO<float> fifo(1024);
+    ASSERT_EQ(1024, fifo.writeAvailable());
+    ASSERT_EQ(100, fifo.write(data.data(), 100));
+    ASSERT_EQ(924, fifo.writeAvailable());
```

Comment:
would an `EXPECT_EQ` fit more or do only asserts make sense here?

Reply:
I didn’t check the style of the other tests but I am used to using ASSERT_xxx in cases like this. The condicion really has to hold, continuing is useless, things are very wrong if any of these conditions fail and the remainder of the test wont make sense.

Reply:
okay, then it fully makes sense to have asserts. Thanks for the clarification.